app/lib/actions/proxy_action.rb

```diff
@@ -93,15 +93,8 @@ def proxy_input(task_id = task.id)
 
     def check_task_status
       response = proxy.status_of_task(proxy_task_id)
-      if %w[stopped paused].include? response['state']
-        if response['result'] == 'error'
-          raise ::Foreman::Exception, _('The smart proxy task %s failed.') % proxy_task_id
```

Comment:
Where this will be checked instead? Or we don't need this?

Reply:
I might have taken a shortcut here that I can't really take.

There's the ::Actions::ProxyAction in here, which (as far as I know) is only used in remote execution. Remote execution has a subclass which inherits from ::Actions::ProxyAction and adds a bunch of extensions. The changes here rely on those extensions. Thank you for pointing this out

Reply:
I kept it mostly intact here and extracted the rex specific parts to https://github.com/theforeman/foreman_remote_execution/pull/900